### DISCUSSION TITLE
Topic partition is not isolated

### DISCUSSION THREAD

**Discussion by hmProgramer** [strimzi/strimzi-kafka-operator#6276] [Q&A]
I found a problem here. I created a topic name test topic partition 9 and copy 3 in one cluster(has 3 broker)

Then I create a topic with the same name  on another cluster(has 7broker), but the partition is 21 and the copy is 3,

Then i  check the first cluster and it becomes that the partition is 21 and the replica is 3? What's going on?

the picture as below:
![Uploading image.png…]()

**Comment by hmProgramer** [comment]
the detail messge as below：
[root@Centos64 kafka_2.12-2.2.0]# bin/kafka-topics.sh --bootstrap-server broker-4309-0.strimzi.kafka.mid:443,broker-4309-1.strimzi.kafka.mid:443,broker-4309-2.strimzi.kafka.mid:443 --topic hm-topic --describe  --command-config /usr/local/kafka_2.12-2.2.0/jks/4309.conf
Topic:hm-topic  PartitionCount:21       ReplicationFactor:3     Configs:message.format.version=2.7-IV2
        Topic: hm-topic Partition: 0    Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic Partition: 1    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
        Topic: hm-topic Partition: 2    Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic Partition: 3    Leader: 1       Replicas: 1,2,0 Isr: 1,2,0
        Topic: hm-topic Partition: 4    Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
        Topic: hm-topic Partition: 5    Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
        Topic: hm-topic Partition: 6    Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic Partition: 7    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
        Topic: hm-topic Partition: 8    Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic Partition: 9    Leader: 1       Replicas: 1,2,0 Isr: 1,2,0
        Topic: hm-topic Partition: 10   Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
        Topic: hm-topic Partition: 11   Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
        Topic: hm-topic Partition: 12   Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic Partition: 13   Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic Partition: 14   Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
        Topic: hm-topic Partition: 15   Leader: 1       Replicas: 1,2,0 Isr: 1,2,0
        Topic: hm-topic Partition: 16   Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
        Topic: hm-topic Partition: 17   Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
        Topic: hm-topic Partition: 18   Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic Partition: 19   Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic Partition: 20   Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
[root@Centos64 kafka_2.12-2.2.0]# clear
[root@Centos64 kafka_2.12-2.2.0]# 
[root@Centos64 kafka_2.12-2.2.0]# 
[root@Centos64 kafka_2.12-2.2.0]# bin/kafka-topics.sh --bootstrap-server broker-4310-0.strimzi.kafka.mid:443,broker-4310-1.strimzi.kafka.mid:443,broker-4310-2.strimzi.kafka.mid:443,broker-4310-3.strimzi.kafka.mid:443,broker-4310-4.strimzi.kafka.mid:443,broker-4310-5.strimzi.kafka.mid:443,broker-4310-6.strimzi.kafka.mid:443 --topic hm-topic --describe  --command-config /usr/local/kafka_2.12-2.2.0/jks/4310.conf
Topic:hm-topic  PartitionCount:21       ReplicationFactor:3     Configs:message.format.version=2.7-IV2
        Topic: hm-topic Partition: 0    Leader: 5       Replicas: 5,3,0 Isr: 5,3,0
        Topic: hm-topic Partition: 1    Leader: 1       Replicas: 1,0,5 Isr: 1,0,5
        Topic: hm-topic Partition: 2    Leader: 4       Replicas: 4,5,1 Isr: 4,5,1
        Topic: hm-topic Partition: 3    Leader: 6       Replicas: 6,1,4 Isr: 6,1,4
        Topic: hm-topic Partition: 4    Leader: 2       Replicas: 2,4,6 Isr: 2,4,6
        Topic: hm-topic Partition: 5    Leader: 3       Replicas: 3,6,2 Isr: 3,6,2
        Topic: hm-topic Partition: 6    Leader: 0       Replicas: 0,2,3 Isr: 0,2,3
        Topic: hm-topic Partition: 7    Leader: 5       Replicas: 5,0,1 Isr: 5,0,1
        Topic: hm-topic Partition: 8    Leader: 1       Replicas: 1,5,4 Isr: 1,5,4
        Topic: hm-topic Partition: 9    Leader: 4       Replicas: 4,1,6 Isr: 4,1,6
        Topic: hm-topic Partition: 10   Leader: 6       Replicas: 6,4,2 Isr: 6,4,2
        Topic: hm-topic Partition: 11   Leader: 2       Replicas: 2,6,3 Isr: 2,6,3
        Topic: hm-topic Partition: 12   Leader: 3       Replicas: 3,2,0 Isr: 3,2,0
        Topic: hm-topic Partition: 13   Leader: 0       Replicas: 0,3,5 Isr: 0,3,5
        Topic: hm-topic Partition: 14   Leader: 5       Replicas: 5,1,4 Isr: 5,1,4
        Topic: hm-topic Partition: 15   Leader: 1       Replicas: 1,4,6 Isr: 1,4,6
        Topic: hm-topic Partition: 16   Leader: 4       Replicas: 4,6,2 Isr: 4,6,2
        Topic: hm-topic Partition: 17   Leader: 6       Replicas: 6,2,3 Isr: 6,2,3
        Topic: hm-topic Partition: 18   Leader: 2       Replicas: 2,3,0 Isr: 2,3,0
        Topic: hm-topic Partition: 19   Leader: 3       Replicas: 3,0,5 Isr: 3,0,5
        Topic: hm-topic Partition: 20   Leader: 0       Replicas: 0,5,1 Isr: 0,5,1

**Comment by scholzj** [comment]
You will have to describe your exact configuration and the steps to reproduce. From the above, I do not think it is really clear what have you actually done and how to reproduce it.

**Reply by hmProgramer** [reply]
i use the k8s strimzi kafka， the  external expose was ingress，and the kafka version is 2.7，protocol is SSL,  I create topic by kafka admin api or the script will reproduce the result。
